pkg/cmd/infra/router/router.go

```diff
@@ -27,28 +28,104 @@ This command launches a router connected to your cluster master. The router list
 created by users and keeps a local router configuration up to date with those changes.`
 )
 
-type templateRouterConfig struct {
-	Config             *clientcmd.Config
-	TemplateFile       string
-	ReloadScript       string
+// routerConfig represents a container for all other router config types.
+// Any global config belongs on this level.
+type routerConfig struct {
+	// Config holds global flags that are passed to all client commands that need
+	// to connect to master.
+	Config *clientcmd.Config
+
+	// Type is the type of plugin being created: f5 or template.
+	Type string
```

Comment:
I'm having difficulty understanding how this would work.  My thinking goes as follows:

1. In `NewCommandRouter`, declare a temporary variable `foo` before the Cobra `Run` callback, so that it will close over the variable.

2. Use `flag.StringVar(type, "type", ...` instead of using `cfg.Type`.

3. Defer the other flag bindings until we know what the type flag's value is.

Is Step 3 possible with Cobra? We are maybe binding flags to things that may or may not exist, with those things' existences' being contingent on the value of the type flag.  How do we conditionalise some flag bindings on the value of another flag?